src/services/codefixes/helpers.ts

```diff
@@ -222,32 +224,17 @@ namespace ts.codefix {
             parameters.push(restParameter);
         }
 
-        return createStubbedMethod(
-            modifiers,
-            name,
-            optional,
-            /*typeParameters*/ undefined,
-            parameters,
-            /*returnType*/ undefined);
-    }
-
-    export function createStubbedMethod(
-        modifiers: ReadonlyArray<Modifier>,
-        name: PropertyName,
-        optional: boolean,
-        typeParameters: ReadonlyArray<TypeParameterDeclaration> | undefined,
-        parameters: ReadonlyArray<ParameterDeclaration>,
-        returnType: TypeNode | undefined) {
         return createMethod(
             /*decorators*/ undefined,
             modifiers,
             /*asteriskToken*/ undefined,
             name,
             optional ? createToken(SyntaxKind.QuestionToken) : undefined,
-            typeParameters,
+            /*typeParameters*/ undefined,
             parameters,
-            returnType,
-            createStubbedMethodBody());
+            /*returnType*/ undefined,

```

Comment:
We might as well give them the return type from the signature, right?

Reply:
Technically it's unsound to union the return types in most cases, but we could do that.

Reply:
Intersection instead? It's better that they get an error when trying to return a value from the method, than that they get an error that the class failed to implement the interface, which are always hard to read..

Reply:
The intersection will be impossible to fulfill most of the time.

```ts
function foo(x: number): number;
function foo(x: string): string;
function foo(x: string | number): ??? {
  // An intersection is impossible to return
}
```

All I'm saying is that it's unsound to do the union, but we do allow it because it's just convenient. You're right that we should probably still do it. Just felt weird.